multimod/README.md

```diff
@@ -0,0 +1,151 @@
+
+An example versioning file is given in [the versions-example.yaml
+file](./docs/versions-example.yaml).
+
+## Creating the app binary
+
+TODO: switch to automatically pulling newest version of `multimod` app binary.
+
+This section describes the current process used for building the Cobra app
+binary, but the process will be updated soon to automatically fetch the most
+recent version of the app.
+
+To build the binary, simply use the following commands:
+
+```sh
+# from the opentelemetry-go-build-tools repo root
+cd multimod
+go build -o multimod main.go
+```
+
+## Verify Module Versioning
+
+Once changes have been completed to the `versions.yaml` file, and the `multimod`
+file is within the target repo, verify that the versioning is correct by running
+the `verify` subcommand:
+
+```sh
+# within the target repo
+./multimod verify
+```
+
+* The script is called with the following parameters:
+  * **versioning-file (optional):** Path to versioning file that contains
+    definitions of all module sets. If unspecified, will default to
+    \<RepoRoot\>/versions.yaml.
+* The following verifications are performed:
+  * `verifyAllModulesInSet` checks that every module (as defined by a `go.mod`
+      file) is contained in exactly one module set.
+  * `verifyVersions` checks that module set version conform to semver semantics
+      and checks that no more than one module set exists for any given non-zero
+      major version.
+  * `verifyDependencies` checks if any stable modules depend on unstable
+    modules.
+    * The stability of a given module is defined by its version in the
+      `version.yaml` file (`v1` and above is stable, `v0` is unstable).
+    * A dependency is defined by the "require" section of the module's `go.mod`
+      file (in the current branch).
+    * A warning will be printed for each dependency of a stable module on an
+      unstable module.
+
+## Prepare a pre-release commit
+
+Update `go.mod` for all modules to depend on the specified module set's new
+release.
+
+1. Run the pre-release script. It creates a branch
+   `pre_release_<module set name>_<new version>` that will contain all release changes.
+
+    ```sh
+    ./multimod prerelease --module-set-name <name>
+    ```
+
+    * The script is called with the following parameters:
+        * **module-set-name (required):** Name of module set whose version is
+          being changed. Must be listed in the module set versioning YAML.
+        * **versioning-file (optional):** Path to versioning file that contains
+          definitions of all module sets. If unspecified will default to
+          (RepoRoot)/versions.yaml.
+        * **from-existing-branch (optional):** Name of existing branch from
+          which to base the pre-release branch. If unspecified, defaults to
+          current branch.
+        * **skip-make (boolean flag):** Specify this flag to skip the 'make
+          lint' and 'make ci' steps. To be used for debugging purposes. Should
```

Comment:
This concept is heavily assumes these targets will exists or even that a project will use make. I'm not sure this a valid assumption to make for a generalized tool we plan to share.